Test/Test/ExtensionTests.cs

```diff
@@ -0,0 +1,149 @@
+            var gv = new { id = idval, partitionkey = pkval }.ToGremlinVertex();
+            Assert.AreEqual(pkval, gv.GetVertexProperties("partitionKey").FirstOrDefault().Value);
+
+            gv = new { Id = idval, partitionKey = pkval }.ToGremlinVertex();
+            Assert.AreEqual(pkval, gv.GetVertexProperties("partitionKey").FirstOrDefault().Value);
+
+            gv = new { ID = idval, PartitionKey = pkval }.ToGremlinVertex();
+            Assert.AreEqual(pkval, gv.GetVertexProperties("partitionKey").FirstOrDefault().Value);
+
+            gv = new { ID = idval, PARTITIONKEY = pkval }.ToGremlinVertex();
+            Assert.AreEqual(pkval, gv.GetVertexProperties("partitionKey").FirstOrDefault().Value);
+        }
+
+        [TestMethod]
+        //we expect the GremlinVertex Label property to be the type name of the object we're converting
+        public void LabelTest()
+        {
+            var obj = new { id = "id", partitionKey = "pk" };
+            var labelval = obj.GetType().Name;
+
+            var gv = obj.ToGremlinVertex();
+            Assert.AreEqual(labelval, gv.Label);
+        }
+
+        [TestMethod]
+        public void SettingIdPkLabelTest()
+        {
+            var idval = "id";
+            var pkval = "pk";
+            var labelval = "label";
+
+            var obj = new { identifier = idval, pk = pkval };
+
+            var gv = obj.ToGremlinVertex("identifier", "pk", labelval);
+            Assert.AreEqual(idval, gv.Id);
+            Assert.AreEqual(labelval, gv.Label);
+            Assert.AreEqual(pkval, gv.GetVertexProperties("partitionKey").FirstOrDefault().Value);
+        }
+
+        [TestMethod]
+        public void Test()
+        {
+            var idval = "id";
+            var pkval = "pk";
+            var nbr = 1;
+            var str = "string";
+            var dte = DateTime.UtcNow;
+
+            var obj = new { id = idval, partitionKey = pkval, nbr, str, dte, NestedProp = new { someprop = "someval" } };
+            var labelval = obj.GetType().Name;
+
+            var gv = obj.ToGremlinVertex();
+            Assert.AreEqual(idval, gv.Id);
+            Assert.AreEqual(labelval, gv.Label);
+
+            var props = gv.GetVertexProperties();
+
+            Assert.IsTrue(props.Count<GremlinVertexProperty>() == 5);
+            Assert.AreEqual("partitionKey", props.FirstOrDefault().Key);
+            Assert.AreEqual(pkval, props.FirstOrDefault().Value);
+            Assert.AreEqual(nbr, props.ElementAt(1).Value);
+            Assert.AreEqual(str, props.ElementAt(2).Value);
+            Assert.AreEqual(dte, props.ElementAt(3).Value);
+
+            var nestedProp = props.ElementAt(4).Value;
+            Assert.AreEqual("someval", nestedProp.GetType().GetProperty("someprop").GetValue(nestedProp));
+        }
+
+        internal class Person
+        {
+            public string Name => "person";
+            public int Age => 10;
+        }
+
+        [TestMethod]
+        public void Test2()
```

Comment:
Maybe a better name here for the test?

Reply:
In general I like to have my test method names contain more information instead of relying on comments. The reason for this is when a test then fails (in VS or in CICD) you immediately know what the test was trying to do without needing to look it up and read a comment. 

For example, the convention i like to use is: `MethodUnderTest_State_Result`. So instead of the name `LabelTest` above, you could do `ToGremlinVertex_NoLabel_SetToType` or something like that.

Its definitely more verbose but I really find it helpful, especially when you have thousands of unit tests. Up to you though!

Reply:
I was going with a Test class for all ToGremlinVertex tests. 
But agree, Test2 is a terrible name which i did late at night and as i was typing it i felt dirty doing it 😆